### DISCUSSION TITLE
Can't preselect "Archive" value on bookmarks

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#24906] [Feature Requests]
### Describe the Improvement

When creating a bookmark we can't set the "archive" value. As the default is to `Show active items` it's not possible to create a direct bookmark for archived-only items.

For sure this can be workaround by disabling the Archive-App-Filter for the collection, but I think this could be optimized.

**Comment by Jopgood** [comment]
I was looking into this one and found:
- Collection Archive information is stored in:
  - `collection.meta` where we can find archive values, what field is the archiver etc.
- the visual Archive filtering is done through URL params
  - As you flick between different archive views, the URL param changes and the archive/not archived data is retrieved based on a filter


We have a couple options with this the way I see it.

We can a) remodel the way archive information is retrieved and use the `stores/presets.ts` to get/set the values but this might be a lot of work

b) we could build a workaround whereby the 'getting' of archive filter could be controlled through `stores/presets.ts` and `use-presets.ts` where we call on the collections store to retrieve the meta data on what field is the archive field etc.

This way you could add this archive meta information to the preset object and subsequently pull that information when generating a bookmark or clicking through different layouts/collections

**Comment by yfi** [comment]
I could really use this too. In the meantime here's a simple hook/hack to provide that functionality.

Your preset could have a redundant condition that is highly unlikely to be used in any real query but identifies it as coming from a preset. Such as `[id][_nnull]=true`. 

If the hook finds this condition in the query, it removes the auto appeneded `[status][_neq]=archived`
condition.

ie.

```js
import { defineHook } from "@directus/extensions-sdk"
import { flatten } from "flat"

export default defineHook(({ filter }) => {
  filter("collection.items.query", (query, meta, data) => {

    const hasPresetFlag = (obj, pathToFind = "id._nnull") => {
      const flat_keys = Object.keys(flatten(obj))
      return !!flat_keys.find((key) => key.includes(pathToFind))
    }

    if (hasPresetFlag(query)) {
      query?.filter?._and?.pop()
    }

    return query

  })
})
```

not ideal, but does the trick for now.
